.gitignore

```diff
@@ -4,3 +4,5 @@
 build
 doc/gh-pages
 Makefile
+.idea
+cmake-build-debug/
```

Comment:
Are these exhaustive? For example, are there other build directory names that make sense? Then we could use `cmake-build-*` for more flexibility.

Reply:
added